### PR TITLE
doc: add acl user guide

### LINKED ISSUE #4409: POSIX ACL
**Why is this needed**:
support [POSIX ACL](https://www.usenix.org/legacy/publications/library/proceedings/usenix03/tech/freenix03/full_papers/gruenbacher/gruenbacher_html/main.html)

- link: #2559 
- link: #512 

---
**What would you like to be added**:
# basic ACL functionality
## reference
- https://github.com/torvalds/linux/commit/60bcc88ad185d512f5718f2f8dcccb483ea8fb73

## what to be added
- add enable-acl flag (also implies default_permissions) to check access acl
  - TODO: For SDK users, user-space ACL checking is required.
- update acl: in setxattr
- default acl: inheritance of default ACLs when new filesystem nodes are
created
- sync between file mode and acl: setattr and setxattr

# compatibility
### why is this needed
- Allowing client mount without ACL will disrupt the synchronization of file mode and ACL.
- Disabling ACL requires a time-consuming conversion of the group class part of file modes for all files (from mask to owner group), so currently, disabling ACL is not supported. 

### what to be added
- add enable-acl flag to meta format
- set fuse.MountOptions
  - DisableXAttrs = !(enable-acl || enable-xattr)
  - EnableAcl = enable-acl
  - IgnoreSecurityLabels = !enable-acl

### why is this needed
If an older version client mounts a newer version of meta format, it will disrupt the synchronization of mode and ACL.

### what to be added
Setting the matching versions for the client and meta.

### cases
1. old client <-> new fs without enable-acl, supported
2. old client <-> new fs with enable-acl, not supported
- logic of "setattr, getattr, mknod" is not compatible
- we need to change format's MinClientVersion when config "enable-acl" to true.
4. new client <-> old fs, supported.  
- two new fields will be added to attr. SQL will sync the table, and KV storage is in []byte format, will ensuring compatibility in encode and decode.
- also support "enable-acl" for old fs.

# storage
## what to be added
- Implement an ACL value sharing mechanism
- We assume that the number of ACLs should be very limited. Having too many ACLs may indicate issues in the way they are being used, and it may be necessary to optimize them through group settings.

1. limited acls (ex. 65536)
- Adding two uint32 variables in the attr to store the index of access ACL and default ACL, enabling quick determination of whether a file/dir has ACL set.
- ACL value is stored in meta with the key 'acl_[index]'.
  - index will keep increasing
  - unused index will not be removed.
- Cache all ACL value in memory

# test
- [posix](https://juicefs.com/docs/zh/community/posix_compatibility/)

### PR DESCRIPTION
**Pull request by jiefenghuang**
close #4409

### CODE REVIEW

File: docs/en/security/posix_acl.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+:::
+
+## Others
+
+- Enabling ACL increases the minimum client version requirement to v1.2.
+- Enabling ACL may have additional performance implications.
+For scenarios with infrequent ACL changes,
+the impact is minimal with memory cache optimization.
+- Enabling ACL will activate extended attributes (xattr) functionality.
+- Enabling ACL is recommended for using ["Sync Accounts between Multiple Hosts"](../administration/sync_accounts_between_multiple_hosts.md)
```
Comment (davies) [inline docs/en/security/posix_acl.md:52]: 这两个不相关

File: docs/en/security/posix_acl.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+---
+
+# POSIX ACL
+
+Version 1.2 supports POSIX ACL. For detailed rules, please refer to:
+
+- [POSIX Access Control Lists on Linux](https://www.usenix.org/legacy/publications/library/proceedings/usenix03/tech/freenix03/full_papers/gruenbacher/gruenbacher_html/main.html#:~:text=Access%20Check%20Algorithm&text=The%20ACL%20entries%20are%20looked,matching%20entry%20contains%20sufficient%20permissions.)
+- [setfacl](https://linux.die.net/man/1/setfacl)
+
+## Usage
```
Comment (davies) [inline docs/en/security/posix_acl.md:12]: Add an example to enable ACL for volume created by old versions.

how to enable it:
1. upgrade all client to 1.2 and remount it
2. juicefs config --enable cli

File: docs/zh_cn/security/posix_acl.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
+---
+sidebar_position: 3
+---
+# POSIX ACL
+
+1.2 版本支持 POSIX ACL, 详细规则参考
+
+- [POSIX Access Control Lists on Linux](https://www.usenix.org/legacy/publications/library/proceedings/usenix03/tech/freenix03/full_papers/gruenbacher/gruenbacher_html/main.html#:~:text=Access%20Check%20Algorithm&text=The%20ACL%20entries%20are%20looked,matching%20entry%20contains%20sufficient%20permissions.)
+- [setfacl](https://linux.die.net/man/1/setfacl)
```
Comment (davies) [inline docs/zh_cn/security/posix_acl.md:9]: 有没有这个文档的中文版？

File: docs/en/security/posix_acl.md (hunk 1 above)
Comment (jiefenghuang) [reply docs/en/security/posix_acl.md:52]: removed

File: docs/zh_cn/security/posix_acl.md (hunk 3 above)
Comment (jiefenghuang) [reply docs/zh_cn/security/posix_acl.md:9]: 这个没有

File: docs/en/security/posix_acl.md (hunk 2 above)
Comment (jiefenghuang) [reply docs/en/security/posix_acl.md:12]: done

**Pull request merged**